src/print_graph.cc

```diff
@@ -34,7 +34,19 @@ namespace detail {
 		}
 	}
 
-	void format_requirements(std::string& label, const task& tsk, subrange<3> execution_range, access_mode reduction_init_mode, const reduction_manager& rm) {
+	std::string get_buffer_label(const buffer_manager* bm, const buffer_id bid) {
+		// if there is no buffer manager or no name defined, the name will be the buffer id.
+		// if there is a name we want "id name"
+		std::string name{};
+		const std::string bid_str = fmt::format("B{}", bid);
+		if(bm != nullptr) { name = bm->get_debug_name(bid); }
+		if(!name.empty()) { name = fmt::format(" \"{}\"", name); }
+		name = fmt::format("{}{}", bid_str, name);
+		return name;
```

Comment:
I find the data flow around `name` a bit hard to follow. Suggestion:

```suggestion
		std::string name;
		if(bm != nullptr) { name = bm->get_debug_name(bid); /* empty if no name is set */ }
		return !name.empty() ? fmt::format("B{} \"{}\"", bid, name) : fmt::format("B{}", bid);
```

Reply:
This changes the output to always include quotes (even if the name is empty). That'd be fine with me though. Maybe even initialize it with `std::string name = "(unnamed)"`?

Reply:
Nevermind the first part, didn't see the ternary operator.